docs/book/src/topics/workload-identity.md

```diff
@@ -0,0 +1,172 @@
+  If you do not have `clusterctl` installed, then follow this [link](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl)
+  to install.
+```bash
+$ clusterctl init --infrastructure azure
+```
+
+### Creating A Workload Cluster
+
+- As mentioned earlier in the doc, cloud provider integration is worked upon
+  and at this stage we need to create an user assigned identity which cloud
+  provider azure can use for authentication. Below are the steps:
+  - [Create](https://learn.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azp#create-a-user-assigned-managed-identity)
+a user-assigned managed identity in Azure.
+  - [Create a role assignment](https://learn.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/howto-assign-access-portal#use-azure-rbac-to-assign-a-managed-identity-access-to-another-resource)
+to give the identity Contributor access to the Azure subscription where the workload cluster will be created.
+
+- Before we can generate a workload cluster YAML configuration set the
+  following environment variables.
+```bash
+export AZURE_SUBSCRIPTION_ID=<your-azure-subscription-id>
+export AZURE_CLIENT_ID=<your-azure-client-id>
+export AZURE_TENANT_ID=<your-azure-tenant-id>
+export AZURE_CONTROL_PLANE_MACHINE_TYPE="Standard_B2s"
+export AZURE_NODE_MACHINE_TYPE="Standard_B2s"
+export AZURE_LOCATION="eastus"
+
+# Identity secret. Though these are not used in workload identity, we still
+# need to set them for the sake of generating the workload cluster YAML configuration
+export AZURE_CLUSTER_IDENTITY_SECRET_NAME="cluster-identity-secret"
+export CLUSTER_IDENTITY_NAME="cluster-identity"
+export AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE="default"
+```
+- Generate a workload cluster template using the following command.
+
+```bash
+$ clusterctl generate cluster azwi-quickstart --kubernetes-version v1.27.3  --worker-machine-count=3 >azwi-quickstart.yaml
+```
+
+- Edit the generated `azwi-quickstart.yaml` to make following changes for
+  workload identity in `AzureClusterIdentity` object.
+  - Change the type to `WorkloadIdentity`.
+  - Remove the `clientSecret` spec.
+
+The AzureClusterIdentity specification should like the following.
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureClusterIdentity
+metadata:
+  name: cluster-identity
+spec:
+  type: WorkloadIdentity
+  allowedNamespaces: {}
+  tenantID: <your-tenant-id>
+  clientID: <your-client-id>
+```
+
+- Change the `AzureMachineTemplate` to include user-assigned-identity by
+  adding the following in its `spec`.
+```yaml
+      identity: UserAssigned
+      userAssignedIdentities:
+      - providerID: /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${USER_ASSIGNED_IDENTITY_NAME}
```

Comment:
A note should be added mentioning that a user-assigned identity needs to be created regardless of whether you're using it to create the federated identity or not, as well as giving it access to the subscription scope.

Reply:
Added it to the top